frameworks/moveos-stdlib/sources/event_queue.move

```
@@ -0,0 +1,217 @@
+        object::borrow_mut_field(event_queue_obj, SUBSCRIBERS_KEY)
+    }
+
+    #[private_generics(E)]
+    public fun emit<E : copy + drop + store>(name: String, event: E) {
+        let event_queue_obj = event_queue<E>(name);
+        let subscribers = borrow_subscribers(event_queue_obj);
+        //We only write the event to the event queue when there are subscribers
+        if(vector::is_empty(subscribers)){
+            return
+        };
+        let head_sequence_number = object::borrow(event_queue_obj).head_sequence_number;
+        let now = timestamp::now_milliseconds();
+        let on_chain_event = OnChainEvent {
+            event: event,
+            emit_time: now,
+        };
+        object::add_field(event_queue_obj, head_sequence_number, on_chain_event);
+        object::borrow_mut(event_queue_obj).head_sequence_number = head_sequence_number + 1;
+        remove_expired_events(event_queue_obj);    
+    }
+
+    public fun consume<E: copy + drop + store>(subscriber_obj: &mut Object<Subscriber<E>>) : Option<E> {
+        let subscriber = object::borrow_mut(subscriber_obj);
+        let subscriber_sequence_number = subscriber.sequence_number;
+        let event_queue_obj = event_queue<E>(subscriber.queue_name);
+        let head_sequence_number = object::borrow(event_queue_obj).head_sequence_number;
+        let tail_sequence_number = object::borrow(event_queue_obj).tail_sequence_number;
+        
+        if (subscriber_sequence_number >= head_sequence_number) {
+            return option::none()
+        };
+        //If the subscriber sequence number is less than the tail sequence number,
+        //It means the subscriber is too slow to consume the event
+        //We should update the subscriber sequence number to the tail sequence number
+        subscriber_sequence_number = if (tail_sequence_number > subscriber_sequence_number) {
+            tail_sequence_number
```

I feel that it is not appropriate to jump directly to the end of the queue here, which may cause inconsistent data on the consumer side. It feels better to return an error.

---

How does the subscriber handle the error?